lib/breadcrumbs.ts

```diff
@@ -0,0 +1,42 @@
+import { convertGithubUrlToShorthand } from './convertGithubUrlToShorthand';
+import { ViewMode } from './enums';
+import { getValidUrlFromInput } from './getValidUrlFromInput';
+import { paramsFromUrl } from './paramsFromUrl';
+import { IssueData } from './types';
+
+interface CrumbData {
+  url: null | string;
+  title: string;
+}
+
+export function getCrumbStringFromIssueData(issue: Pick<IssueData, 'html_url' | 'title'>): string {
+  return `${convertGithubUrlToShorthand(new URL(issue.html_url))}@@${issue.title}`
+}
+
+/**
+ *
+ * @param crumbs A string of decoded crumbs. Example: `owner/repo#1@@Issue 1,owner/repo#2@@Issue 2`
+ * @param viewMode the current view mode, to ensure links stay in the current view
+ * @returns
+ */
+export function getCrumbDataFromCrumbString(crumbs: string, viewMode: ViewMode): CrumbData[] {
+  const crumbItems = crumbs.split(',');
+  const crumbData: CrumbData[] = [];
+  crumbItems.forEach((crumb, index) => {
+    const crumbParts = crumb.split('@@');
+    const { owner, repo, issue_number } = paramsFromUrl(getValidUrlFromInput(crumbParts[0]).toString());
+    const url = new URL(`/roadmap/github.com/${owner}/${repo}/issues/${issue_number}#${viewMode}`, window.location.origin);
+    const crumbsForUrl = crumbItems.slice(0, index).join(',');
+
+    if (crumbsForUrl.length > 0) {
+      url.searchParams.append('crumbs', crumbsForUrl)
+    }
+
+    crumbData.push({
+      url: url.toString(),
+      title: crumbParts[1],
+    });
+  });
+
```

Comment:
I thought this was a pretty uncreative and novel solution. It's a fairly idiomatic pattern (other than the choice of `@@` for a separator) across websites doing custom query parameter parsing. 

I considered JSON encoding data and decided to start with something resulting in a much smaller url. Explicitly named keys, fields, and syntax characters for JSON-encoded data in the URL would larger. We could do protobuf and encoding of data to reduce that size, but I feel like that's a much more complicated solution than we could get done before release. It would also be much more difficult to maintain than a simple `value@@value` convention.

I'm open to a simple solution that can get us across the finish line and doesn't easily break on deeply nested roadmaps. I'll explore JSON stringifying the url+titles in a sorted array now that things are working and see how it fares.

Reply:
I did json parsing in #229 . I don't agree that it's simpler, but I'm willing to do that if you think it's better. Either way, that PR is where I also improved things significantly, so I would like to merge that PR into this one prior to merging to live